### DISCUSSION TITLE
URL acquisition after file upload

### DISCUSSION THREAD

**Discussion by sdr0x07b6** [supabase/supabase#18127] [Feature Requests]
We request a specification that returns a URL as the return value of the `upload()` method.

In many cases, we would like to know the URL when the upload to storage is completed, so we feel that it is an unnatural specification not to have it as the return value of the `upload()` method. Is this specification intentionally made in this way?

We need to use the `getPublicUrl()` method to get the URL after the upload is complete, but it is tedious each time, requires two requests each time and takes time.

The path is returned from the `upload()` method, but the path is what you specify yourself when uploading and you don't need it to be returned. What you want is the URL.

**Comment by GaryAustin1** [comment]
getPublicUrl does NOT do a server call or anything on the net.   It is simply a string concatenation function.  So the value of not using it and get a url back is minimal as far as performance.

**Reply by sdr0x07b6** [reply]
That was a relief. thank you for teaching me.
However, I think it would still be more convenient if `upload()` returned the URL directly. Is there a rational reason for not returning it?

I understand that there is no request load. However, I think there is a big advantage to having `upload()` return a URL, and I would like to see it implemented even more if it can be done by simply concatenating strings.

The return value of `upload()` is currently `{ path: '...' }`. If you extend it like `{ path: '...', url: '...' }`, it will not affect existing applications, so please consider it.

**Reply by GaryAustin1** [reply]
Your feature request is still open.  I am not an employee and just letting you know that for now (or ever if not implemented) it is just a bit of code overhead versus performance delay.
